dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/SingularityIT.java

```diff
@@ -0,0 +1,140 @@
+
+    @Before
+    @Override
+    public void resetDBBetweenTests() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
+    }
+
+    /**
+     * Tests that a simple CWL workflow can be run in Singularity instead of Docker
+     */
+    @Test
+    public void runCwlWorkflow() {
+        // manually register the CWL version of the md5sum-checker workflow locally
+        testingPostgres.runUpdateStatement("update enduser set isadmin = 't' where username = 'DockstoreTestUser2';");
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        Workflow workflow = workflowApi.manualRegister(SourceControl.GITHUB.getFriendlyName(), "DockstoreTestUser2/md5sum-checker",
+                "/checker-workflow-wrapping-workflow.cwl", "test", "cwl", null);
+        workflowApi.refresh(workflow.getId());
+
+        // start saving the output so as to make sure that Singularity actually runs instead of Docker
+        PrintStream old = System.out;  // save the regular output stream so it can be reset later
+        TeeOutputStream teeOut = new TeeOutputStream(System.out, outContent);
+        PrintStream out = new PrintStream(teeOut, true);  // also print it to the screen
+        System.setOut(out);
+
+        // run the md5sum-checker workflow
+        Client.main(new String[] {
+            "--config",  // this config file passes the --singularity option to cwltool
+            ResourceHelpers.resourceFilePath("config_for_singularity"),
+            "workflow",
+            "launch",
+            "--entry",
+            SourceControl.GITHUB.toString() + "/DockstoreTestUser2/md5sum-checker/test",
+            "--json", ResourceHelpers.resourceFilePath("md5sum_cwl.json")
+        });
+        System.setOut(old);  // reset the output stream to stdout
+
+        // the message "Creating SIF file" will only be in the output if the Singularity command starts successfully
+        Assert.assertTrue(outContent.toString().contains("Creating SIF file"));
+    }
+
+    /**
+     * Tests that a simple WDL workflow can be run in Singularity instead of Docker
+     */
+    @Test
+    public void runWDLWorkflow() {
+        // manually register the WDL version of the md5sum-checker workflow locally
+        testingPostgres.runUpdateStatement("update enduser set isadmin = 't' where username = 'DockstoreTestUser2';");
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        Workflow workflow = workflowApi.manualRegister(SourceControl.GITHUB.getFriendlyName(), "DockstoreTestUser2/md5sum-checker",
+                "/checker-workflow-wrapping-workflow.wdl", "test", "wdl", null);
+        workflowApi.refresh(workflow.getId());
+
+        // make a tmp copy of the dockstore config and add to it the cromwell conf file option
```

Comment:
It's not clear to me, what location varies?

Reply:
in the dockstore config file for this test, there needs to be a line:
`cromwell-vm-options: -Dconfig.file=<absolute path to cromwell conf file, which tells cromwell to run with singularity instead of docker>`. That absolute path is different if it's running on travis or locally, so in other places, `ResourceHelpers.resourceFilePath()` is used to find it. But since the path needs to be in the config I had to write it to a tmp copy of the config.

Reply:
Ok, thanks. I think this comment can be added to the Javadoc/comment for `generateCromwellConfig`, particularly the fact that the absolute path is needed and changing between travis or locally. Thanks!